### PR TITLE
Add support for Satellite/Spacewalk which have slightly different API

### PR DESCRIPTION
**Pull request by jhutar**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

packaging/os/rhn_register.py
##### Summary:

Spacewalk/Satellite API is a bit different, so to add a support for subscribing into specific channels slightly more code was needed. Tested against Hosted (rhn.redhat.com) and against Satellite.
##### Example:

```
  - name: ensure we are registered to satellite
    rhn_register: state=present server_url=https://satellite.example.com/XMLRPC username=<user> password=<pass> sslcacert=/etc/sysconfig/rhn/RHN-ORG-TRUSTED-SSL-CERT channels=rhel-x86_64-server-6,rhn-tools-rhel-x86_64-server-6
```

or:

```
  - name: ensure we are registered to hosted
    rhn_register: state=present server_url=https://xmlrpc.rhn.redhat.com/XMLRPC username=<user> password=<pass> sslcacert=/usr/share/rhn/RHNS-CA-CERT channels=rhel-x86_64-server-6,rhn-tools-rhel-x86_64-server-6
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @jhutar. To the current maintainers, @barnabycourt, @vritant, @flossware, @alikins please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: packaging/os/rhn_register.py [hunk 1]
```diff
@@ -298,6 +319,16 @@ def _subscribe(self, channels=[]):
                 if re.search(wanted_repo, available_channel):
                     rc, stdout, stderr = self.module.run_command(rhn_channel_cmd + " --add --channel=%s" % available_channel, check_rc=True)
 
+    def _is_hosted(self):
+        '''
+        Return Tru if we are running against Hosted (rhn.redhat.com) or False
```
Comment (alikins) [inline packaging/os/rhn_register.py]: s/Tru/True
s/othervise/otherwise

**Comment by alikins** [comment]
needs_revision (albeit just typos)

File: packaging/os/rhn_register.py [hunk 2]
```diff
@@ -279,10 +279,31 @@ def unregister(self):
...
+                    continue
+                if self.api('channel.software.getDetails', ch)['parent_channel_label'] == '':
+                    new_base = ch
+                else:
+                    if ch not in new_childs:
+                        new_childs.append(ch)
+            out_base = 0
+            out_childs = 0
+            if new_base:
+              out_base = self.api('system.setBaseChannel', self.systemid, new_base)
```
Comment (alikins) [inline packaging/os/rhn_register.py]: should be 4 space indent here

**Comment by gregdek** [comment]
Thanks @jhutar for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jhutar** [comment]
@alikins Oh my, so lame mistakes. Thanks for a review. Should be fixed now.

**Comment by jhutar** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @jhutar. To the current maintainers, @barnabycourt, @vritant, @flossware, @alikins please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by alikins** [comment]
code looks good to me.
needs_revision to squash into a single commit (ala http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes)

**Comment by jhutar** [comment]
Commits squashed. It is ready_for_review now. Thank you!

**Comment by alikins** [comment]
looks good to me.

:shipit: shipit

**Comment by gregdek** [comment]
Thanks again to @jhutar for this PR, and thanks @barnabycourt, @vritant, @flossware, @alikins for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**